### PR TITLE
Update to VecCore integration in ROOT

### PR DESCRIPTION
**Pull request by amadio**
This PR improves VecCore integration into ROOT. Before we copied all VecCore and Vc headers into the build directory, since rootcling did not get the right includes when Vc or VecCore were builtin. Now the right includes are propagates by calling `set_property()` with the necessary include directories. Ideally, in the future we should take the includes not from the `DIRECTORY` property, but from the target dependencies for each subcomponent.

### CODE REVIEW

**Comment by amadio** [comment]
@phsft-bot build also on mac1012/native with flags -Dbuiltin_veccore=ON -Dbuiltin_vc=ON

**Comment by amadio** [comment]
@phsft-bot build also on mac1012/native with flags -Dbuiltin_veccore=ON -Dbuiltin_vc=ON

**Comment by amadio** [comment]
@phsft-bot build on ubuntu16/native with flags -Dbuiltin_vc=ON -Dbuiltin_veccore=ON

**Comment by amadio** [comment]
@phsft-bot build on ubuntu14/native with flags -Dbuiltin_vc=ON -Dbuiltin_veccore=ON

File: hist/hist/inc/TF1.h [hunk 1]
```diff
@@ -21,17 +21,17 @@
 //                                                                      //
 //////////////////////////////////////////////////////////////////////////
 
-#include "RConfigure.h"
```
Comment (pcanal) [inline hist/hist/inc/TF1.h]: Why remove the explicit inclusion of RConfigure?

Comment (amadio) [inline hist/hist/inc/TF1.h]: It's only used to check for `R__HAS_VECCORE`, which is provided by `Math/Math.h`.

**Comment by amadio** [comment]
@pcanal What do you think would be the right place to include VecCore in ROOT? I want to avoid injecting it as a dependency when it's not needed, but that seems to cause problems with the PCH on Mac.

**Comment by amadio** [comment]
@phsft-bot build also on centos7/gcc49, mac1012/native, slc6/gcc62 with flags -Dbuiltin_vc=ON -Dbuiltin_veccore=ON

**Comment by amadio** [comment]
Both Vc and VecCore are now installed into `${CMAKE_BINARY_DIR}/externals` with `${CMAKE_PREFIX_INSTALL}` appended (unfortunately necessary to make Vc work at the end, since Vc is not relocatable). This PR needs https://github.com/root-project/roottest/pull/65 to be merged for tests to pass (that PR properly propagates the new necessary include directories when running tests).

**Comment by amadio** [comment]
@phsft-bot build

**Comment by amadio** [comment]
Will close this PR and open a new one. I think that the dirty build directories are interfering with testing.

**Pull request closed without merging**